File: API/Models/Transaction.cs
﻿using System.ComponentModel.DataAnnotations;

namespace API.Models
{
    public class Transaction
    {
        [Key]
        public int TransactionId { get; set; }
        [Required]
        public int AccountId { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        [RegularExpression(@"^\d+.?\d{0,2}$")]
        public decimal Amount { get; set; }
        [Required]
        public DateTime Date { get; set; }
        [Required]
        public int CustomerId { get; set; }

        public Account Account { get; set; }
        public Customer Customer { get; set; }
    }
}

File: react-app/src/components/interfaces/Transaction.ts
export interface Transaction {
    transactionId: number,
    accountId: number,
    description: string,
    amount: number,
    date: Date | null,
    customerId: number
}

export interface TransactionDto {
    transactionId?: number,
    accountId: number,
    description: string,
    amount: number,
    date: string,
    customerId: number
}

export const emptyTransaction = {
    transactionId: 0,
    accountId: 0,
    description: "",
    amount: 0,
    date: null,
    customerId: 1
}